Formula/p/pkl.rb

```diff
@@ -0,0 +1,37 @@
+class Pkl < Formula
+  version "0.25.2"
+  url "https://github.com/apple/pkl/archive/refs/tags/0.25.2.tar.gz"
+  sha256 "810f6018562ec9b54a43ba3cea472a6d6242e15da15b73a94011e1f8abc34927"
+
+  desc "CLI for the Pkl programming language"
+  homepage "https://pkl-lang.org"
+  license "Apache-2.0"
+
+  env :std
+  depends_on "openjdk@17" => :build
+
+  def install
+    ENV["JAVA_HOME"] = Formula["openjdk@17"].opt_prefix
+    task_os = "mac"
+    bin_os = "macos"
+    arch = Hardware::CPU.arm? ? "Aarch64" : "Amd64"
+
+    if OS.linux?
+      task_os = "linux"
+      bin_os = "linux"
+    end
+
+    job_name = "#{task_os}Executable#{arch}"
+    system "git", "apply", "patches/graalVm23.patch" if OS.mac? && Hardware::CPU.arm?
+    system "./gradlew", "--info", "--stacktrace", "-DreleaseBuild=true", job_name
+    bin.install "pkl-cli/build/executable/pkl-#{bin_os}-#{arch.downcase!}" => "pkl"
```

Comment:
I actually tried this in my local, it did not work.

Reply:
Interesting; it's working on my machine. What error are you getting?

Reply:
relates to `graalVm23`, ENV.cc related

Reply:
Do you have a log that you can share?

Reply:
yeah, I can rebuild again (I am using sonoma arm build)

I was in the middle of troubleshooting the build last time :)

Reply:
here you go

```
> Task :pkl-cli:macExecutableAarch64
WARNING: package com.oracle.truffle.api.impl.asm.tree.analysis not in org.graalvm.truffle
========================================================================================================================
GraalVM Native Image: Generating 'pkl-macos-aarch64' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------

[1/8] Initializing...                                                                                    (0.0s @ 0.15GB)
Error: Unable to detect supported DARWIN native software development toolchain.
Querying with command '/opt/homebrew/Library/Homebrew/shims/mac/super/cc -v' prints:
  cc: The build tool has reset ENV; --env=std required.
Error: To prevent native-toolchain checking provide command-line option -H:-CheckToolchain
com.oracle.svm.core.util.UserError$UserException: Unable to detect supported DARWIN native software development toolchain.
Querying with command '/opt/homebrew/Library/Homebrew/shims/mac/super/cc -v' prints:
  cc: The build tool has reset ENV; --env=std required.
To prevent native-toolchain checking provide command-line option -H:-CheckToolchain
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:126)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.addSkipCheckingInfo(CCompilerInvoker.java:106)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.<init>(CCompilerInvoker.java:74)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker$DarwinCCompilerInvoker.<init>(CCompilerInvoker.java:279)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.create(CCompilerInvoker.java:84)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:929)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:579)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:539)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:408)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:612)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:134)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:94)
------------------------------------------------------------------------------------------------------------------------
                         0.4s (9.9% of total time) in 9 GCs | Peak RSS: 0.54GB | CPU load: 1.85
========================================================================================================================
Finished generating 'pkl-macos-aarch64' in 2.3s.

> Task :pkl-cli:macExecutableAarch64 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':pkl-cli:macExecutableAarch64'.
> Process 'command '/Users/rui/Library/Caches/Homebrew/java_cache/.graalvm/graalvm-jdk-17.0.9_macos-aarch64_bin/Contents/Home/bin/native-image'' finished with non-zero exit value 1
```

Reply:
this is the full build log, https://gist.github.com/chenrui333/e749cf03336a3412459453ac41886265

Reply:
Thanks! I added: `env :std` to the formula: https://github.com/Homebrew/homebrew-core/pull/162857/files#diff-ec86d9164e3c5a94c83213a8ed5a118d10dda4d93fcb14629ef006f21eea63cbR12

Hopefully this fixes it for you.